homeassistant/components/mqtt/climate.py

```diff
@@ -412,6 +456,9 @@ def _setup_from_config(self, config: ConfigType) -> None:
         if self._topic[CONF_TEMP_HIGH_STATE_TOPIC] is None:
             self._attr_target_temperature_high = config[CONF_TEMP_INITIAL]
 
+        if self._topic[CONF_HUMIDITY_STATE_TOPIC] is None:
+            self._attr_target_humidity = config[CONF_HUMIDITY_INITIAL]
+
```

Comment:
```suggestion
```
At `self._attr_target_humidity = None` at self.__init__() instead if `CONF_HUMIDITY_STATE_TOPIC` is `None`